src/kite/base.js

```diff
@@ -63,29 +71,31 @@ class Kite extends Emitter {
 
   setToken(token) {
     // FIXME: this setter is not symettrical with the getter
-    if (this.options.auth && this.options.auth === AuthType.token) {
+    const { auth } = this.options
+
+    if (!auth || auth === AuthType.token) {
```

Comment:
but this new logic is wrong, afais the previous one was doing something completely different; it was checking for if `auth` option is already set and it's `AuthType.token` and throws error in that case, otherwise it will set the `auth.key` to `token` with one mistake which is not checking `auth` existence on `options`. we can check only for that one I think otherwise it's going to throw in any case if there is no `auth` set in `options` which should break `renewToken` flow in `kontrol` afais.